kamon-akka-sharding/src/main/scala/kamon/akka/sharding/instrumentation/AkkaShardingInstrumentation.scala

```diff
@@ -0,0 +1,77 @@
+package akka.kamon.instrumentation
+
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.{ Executors, ThreadFactory, TimeUnit }
+
+import akka.actor.{ ActorRef, ExtendedActorSystem, Props }
+import akka.cluster.sharding.ShardCoordinator.ShardAllocationStrategy
+import akka.cluster.sharding.ShardRegion.{ CurrentShardRegionState, ExtractEntityId, ExtractShardId, GetShardRegionState }
+import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings }
+import akka.pattern._
+import akka.util.Timeout
+import com.typesafe.config.Config
+import kamon.Kamon
+import kamon.akka.sharding.metrics.ShardMetrics
+import kamon.metric.Entity
+import org.aspectj.lang.annotation._
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+object AkkaShardingInstrumentation {
+  implicit lazy val ec: ExecutionContext = {
```

Comment:
I had to create a dedicated `ExecutionContext`, since I could not access the default dispatcher from the Kamon actor system.